src/lib.rs

```diff
@@ -406,6 +416,15 @@ impl<'a> FMatcher<'a> {
                         ptn = &ptn[ptnm.end()..];
                         text = &text[ptnm.start()..];
                         if let Some(textm) = text_re.find(text) {
+                            if self.options.distinct_name_matching {
+                                for (x, y) in names.iter() {
```

Comment:
One option to avoid the linear scan would be to use a bidirectional map, for example: https://crates.io/crates/bimap

As I understand, the scan would become a hashing operation instead.

I dunno if it's worth it. I doubt many tests will have more than 5 names to match...

What do you think?

Reply:
I doubt it's worth it for performance and I'd rather avoid the extra library unless we really need it. We can come back and do this later if it does turn out to be a bottleneck.

Reply:
Another option would be to hold a set of values already bound to names. But that uses memory.

Reply:
That doesn't work, unfortunately, because you need to be able to distinguish all other names from your own.

Reply:
> We can come back and do this later if it does turn out to be a bottleneck

OK